### PR TITLE
Added functions for the new pause feature.

### PR DESCRIPTION
**Pull request by wesdean**
I have added some functions to use the new pause feature that Recurly has recently added to their API.

### CODE REVIEW

File: subscriptions_test.go [hunk 1]
```diff
@@ -1031,3 +1031,43 @@ func TestSubscriptions_Postpone(t *testing.T) {
...
+
+	mux.HandleFunc("/v2/subscriptions/44f83d7cba354d5b84812419f923ea96/pause", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "PUT" {
+			t.Fatalf("unexpected method: %s", r.Method)
+		}
+		w.WriteHeader(200)
+		fmt.Fprint(w, `<?xml version="1.0" encoding="UTF-8"?><subscription></subscription>`)
+	})
+
+	r, _, err := client.Subscriptions.Pause("44f83d7cba354d5b8481-2419f923ea96", 1) // UUID has dashes and should be sanitized
```
Comment (cristiangraz) [inline subscriptions_test.go]: 👍

File: client.go [hunk 2]
```diff
@@ -109,7 +109,7 @@ func (c *Client) newRequest(method string, action string, params Params, body in
 	))
 	req.Header.Set("Authorization", fmt.Sprintf("Basic %s", c.apiKey))
 	req.Header.Set("Accept", "application/xml")
-	req.Header.Set("X-Api-Version", "2.5")
+	req.Header.Set("X-Api-Version", "2.11")
```
Comment (cristiangraz) [inline client.go]: Do you know if there are any other any other API changes between 2.5 and 2.11 that could be breaking?

Comment (wesdean) [reply client.go]: I do not know of any breaking changes. I don't see any notes about breaking changes in the release notes from 2.5 to 2.12.

**Comment by katenicoletti** [comment]
Hi @wesdean! Please add a `HasPausedSubscription` field to the `Account` object and `SubscriptionStatePaused` constant (at the top of `subscriptions.go`). There have been breaking changes between v2.5 and v2.11 that the tests don't show because we mock the api response data, so data structure changes have to be manually updated. I am working on a PR for that right now. Once that's done, I'll ask you to rebase, and we'll be able to merge this in. Thanks!

File: accounts_test.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ func TestAccounts_Encoding(t *testing.T) {
 		{v: recurly.Account{AcceptLanguage: "en_US"}, expected: "<account><accept_language>en_US</accept_language></account>"},
 		{v: recurly.Account{FirstName: "Larry", Address: recurly.Address{Address: "123 Main St.", City: "San Francisco", State: "CA", Zip: "94105", Country: "US"}}, expected: "<account><first_name>Larry</first_name><address><address1>123 Main St.</address1><city>San Francisco</city><state>CA</state><zip>94105</zip><country>US</country></address></account>"},
 		{v: recurly.Account{Code: "test@example.com", BillingInfo: &recurly.Billing{Token: "507c7f79bcf86cd7994f6c0e"}}, expected: "<account><account_code>test@example.com</account_code><billing_info><token_id>507c7f79bcf86cd7994f6c0e</token_id></billing_info></account>"},
+		{v: recurly.Account{HasPausedSubscription: true}, expected: "<account><has_paused_subscription>true</has_paused_subscription></account>"},
```
Comment (katenicoletti) [inline accounts_test.go]: 👍

File: .idea/codeStyles/codeStyleConfig.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (katenicoletti) [inline .idea/codeStyles/codeStyleConfig.xml]: I don't think you meant to commit this :-)

**Comment by katenicoletti** [comment]
Hey @wesdean, we just merged the update for v2.11. You will need to rebase, and update any `reflect.DeepEqual` comparisons to use `cmp.Diff` if that's applicable. Then we'll be good to go here. Thank you!

**Comment by katenicoletti** [comment]
Thanks @wesdean

**Pull request merged**